src/lib/zf/attr.c

```diff
@@ -282,6 +283,7 @@ int zf_attr_set_str(struct zf_attr* attr, const char* name, const char* val)
       return -ENOMSG;
     }
     char** p = get_field_str(attr, f);
+    free(*p);
```

Comment:
It is broken as it currently stands. If zf_log_redirect fails then we have free(*p) but it is not set to NULL.
The original point from previous commit is not addressed (can redirect but strdup fails). Are you saying this doesn't matter because you don't use local attribute name?

Reply:
Correct. Thanks for spotting this  - I've set it to NULL. 

If strdup fails, not only will log use the global value (already set at the point you refer to), but also ENOMEM probably means there is a serious memory shortage hence I would not think an earlier evaluation would make a difference. Moreover, let suppose I do strdup first, if zf_log_redirect fails after strdup has already succeeded, the code must handle the deallocation of tmpdup which is more expensive (considering zf_log_redirect failing is more likely than a memory shortage).